crates/bitcoin-da/tests/verifier.rs

```diff
@@ -0,0 +1,593 @@
+
+            // Malform witness
+            block_txs[0].input[0].witness = Witness::from_slice(&[vec![1; 32]]);
+
+            // Recreate inclusion proof
+            let tree = BitcoinMerkleTree::new(
+                block_txs
+                    .iter()
+                    .map(|t| t.compute_txid().to_raw_hash().to_byte_array())
+                    .collect(),
+            );
+            let mut inclusion_proof = InclusionMultiProof {
+                wtxids: block_txs
+                    .iter()
+                    .map(|t| t.compute_wtxid().to_byte_array())
+                    .collect(),
+                coinbase_tx: block_txs[0].clone(),
+                coinbase_merkle_proof: tree.get_idx_path(0),
+            };
+            // Coinbase tx wtxid should be [0u8;32]
+            inclusion_proof.wtxids[0] = [0; 32];
+
+            assert!(verifier
+                .verify_transactions(
+                    &block.header,
+                    &b_txs,
+                    inclusion_proof,
+                    b_completeness_proof.clone(),
+                    DaNamespace::ToBatchProver,
+                )
+                .is_err());
+        }
+
+        // False coinbase script pubkey should fail
+        {
+            let mut block_txs = block.txdata.clone();
+
+            let idx = block_txs[0]
+                .output
+                .iter()
+                .position(|output| {
+                    output
+                        .script_pubkey
+                        .to_bytes()
+                        .starts_with(WITNESS_COMMITMENT_PREFIX)
+                })
+                .unwrap();
+            // Malform coinbase script pubkey
+            let mut bytes = block_txs[0].output[idx].script_pubkey.to_bytes();
+            bytes[0] = bytes[0].wrapping_add(1);
+
+            block_txs[0].output[idx].script_pubkey = ScriptBuf::from_bytes(bytes);
+
+            // Recreate inclusion proof
+            let tree = BitcoinMerkleTree::new(
+                block_txs
+                    .iter()
+                    .map(|t| t.compute_txid().to_raw_hash().to_byte_array())
+                    .collect(),
+            );
+            let mut inclusion_proof = InclusionMultiProof {
+                wtxids: block_txs
+                    .iter()
+                    .map(|t| t.compute_wtxid().to_byte_array())
+                    .collect(),
+                coinbase_tx: block_txs[0].clone(),
+                coinbase_merkle_proof: tree.get_idx_path(0),
+            };
+            // Coinbase tx wtxid should be [0u8;32]
+            inclusion_proof.wtxids[0] = [0; 32];
+
+            assert!(verifier
+                .verify_transactions(
+                    &block.header,
+                    &b_txs,
+                    inclusion_proof,
+                    b_completeness_proof.clone(),
+                    DaNamespace::ToBatchProver,
+                )
+                .is_err());
```

Comment:
don't you want to assert error types?

Reply:
asserted errors: https://github.com/chainwayxyz/citrea/pull/1515/commits/341755599bebd53a7fa33afeb2d8e458f51a411a